addons/html_editor/static/src/main/link/link_plugin.js

```diff
@@ -273,229 +290,254 @@ export class LinkPlugin extends Plugin {
+            const imageNode = findInSelection(selection, "img");
+            if (imageNode?.parentNode?.tagName === "A" && linksInSelection.length < 2) {
+                this.openLinkTools(imageNode.parentElement);
             } else {
-                this.overlay.close();
+                this.linkInDocument = null;
+                this.closeLinkTools();
             }
         } else {
-            const linkEl = closestElement(selection.anchorNode, "A");
-            if (!linkEl) {
-                this.overlay.close();
-                this.removeCurrentLinkIfEmtpy();
-                return;
-            }
-            if (linkEl !== this.linkElement) {
-                this.removeCurrentLinkIfEmtpy();
-                this.overlay.close();
-                this.linkElement = linkEl;
-            }
-
-            // if the link includes an inline image, we close the previous opened popover to reposition it
-            const imageNode = linkEl.querySelector("img");
-            if (imageNode) {
-                this.removeCurrentLinkIfEmtpy();
-                this.overlay.close();
-            }
-
-            const linkProps = {
-                ...props,
-                isImage: false,
-                linkEl: this.linkElement,
-                onApply: (url, label, classes) => {
-                    this.linkElement.href = url;
-                    if (cleanZWChars(this.linkElement.innerText) === label) {
-                        this.overlay.close();
-                        this.dependencies.selection.setSelection(
-                            this.dependencies.selection.getEditableSelection()
-                        );
-                    } else {
-                        const restore = prepareUpdate(...leftPos(this.linkElement));
-                        this.linkElement.innerText = label;
-                        restore();
-                        this.overlay.close();
-                        this.dependencies.selection.setCursorEnd(this.linkElement);
-                    }
-                    if (classes) {
-                        this.linkElement.className = classes;
-                    } else {
-                        this.linkElement.removeAttribute("class");
-                    }
-                    this.dependencies.selection.focusEditable();
-                    this.removeCurrentLinkIfEmtpy();
-                    this.dependencies.history.addStep();
-                },
-            };
-
-            if (linkEl.isConnected) {
-                // pass the link element to overlay to prevent position change
-                this.overlay.open({ target: this.linkElement, props: linkProps });
+            const closestLinkElement = closestElement(selection.anchorNode, "A");
+            if (closestLinkElement) {
+                if (closestLinkElement !== this.linkInDocument) {
+                    this.openLinkTools(closestLinkElement);
+                }
```

Comment:
I think we want to `openLinkTools` when `closestLinkElement` exists (without the equal check).

For example,
1. if we click a link, the popover opens,
2. then if the cursor moves inside this same link, the popover should stay open (even tho the check succeeds)